sdk/sdktypes/integration_id.go

```diff
@@ -22,8 +23,33 @@ func StrictParseIntegrationID(s string) (IntegrationID, error) { return Strict(P
 
 func IsIntegrationID(s string) bool { return IsIDOf[integrationIDTraits](s) }
 
-func IntegrationIDFromName(name string) IntegrationID {
-	txt := fmt.Sprintf("%s_%026x", integrationIDKind, kittehs.HashString64(name))
+const integrationIDFromNamePrefix = "3kth"
+
+func NewIntegrationIDFromName(name string) IntegrationID {
+	// a hint to the original name is encoded in the ID.
+	idName := strings.Map(func(r rune) rune {
+		switch r {
+		case 'i', 'l':
+			return '1'
+		case 'o':
+			return '0'
+		case 'u':
+			return 'v'
+		default:
+			return r
+		}
+	}, name)
+
+	idName = fmt.Sprintf("%06s", idName)
+	if len(idName) > 6 {
+		idName = idName[:6]
+	}
```

Comment:
Limit the name's length before the character mapping to avoid wasted work.

Reply:
done